tests/jballerina-integration-test/src/test/resources/packaging/listener_stophandler_async_call_test/modules/moduleA/main.bal

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2022 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/jballerina.java;
+import ballerina/lang.runtime;
+
+public function stopHandlerFunc() returns error? {
+    runtime:sleep(1);
+    println("StopHandler1 of moduleA");
+}
+
+public function stopHandlerFunc2() returns error? {
+    runtime:sleep(1);
+    println("StopHandler2 of moduleA");
+}
+
+function init() {
+    runtime:onGracefulStop(stopHandlerFunc);
+    runtime:onGracefulStop(stopHandlerFunc2);
+}
+
+public class ListenerA {
+
+    private string name = "";
+    public function init(string name) {
+        self.name = name;
+    }
+
+    public function 'start() returns error? {
+    }
+
+    public function gracefulStop() returns error? {
+        runtime:sleep(1);
+        if (self.name == "ModA") {
+            panic error("graceful stop of ModuleA");
+        }
+    }
+
+    public function immediateStop() returns error? {
+    }
+
+    public function attach(service object {} s, string[]|string? name = ()) returns error? {
+    }
+
+    public function detach(service object {} s) returns error? {
+    }
+}
+
+public class ListenerB {
+
+    private string name = "";
+    public function init(string name) {
+        self.name = name;
+    }
+
+    public function 'start() returns error? {
+    }
+
+    public function gracefulStop() returns error? {
+        runtime:sleep(2);
+    }
+
+    public function immediateStop() returns error? {
+    }
+
+    public function attach(service object {} s, string[]|string? name = ()) returns error? {
+    }
+
+    public function detach(service object {} s) returns error? {
+    }
+}
+
+listener ListenerA listenerA = new ListenerA("ModA");
+
+final ListenerB listenerB = new ListenerB("ListenerObjectB");
+
+public function main() {
+    final ListenerA listenerA = new ListenerA("ModDynA");
+    runtime:registerListener(listenerA);
+    runtime:registerListener(listenerB);
+
+    checkpanic listenerA.gracefulStop();
+    runtime:deregisterListener(listenerA);
+
+    checkpanic listenerB.gracefulStop();
```

Comment:
why we need to call gracefullStop here. Should it be automatically call when server shutdown?

Reply:
Yes, it will be called during the server shutdown. We don't need to call it manually unless we want to execute that as a normal function call. Removed most of those unnecessary function call lines since they are not actually need to be tested here.